src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,73 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
+    private final int[] array;
+    private final int start;
+    private final int end;
+
+    public MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length - 1);
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start + 1 <= THRESHOLD) {
+            insertionSort(array, start, end);
+        } else {
+            int mid = (start + end) / 2;
```

Comment:
The calculation of `mid` could be improved for clarity by using `int mid = start + (end - start) / 2;`. This avoids potential overflow issues, although it's not a concern with int in Java.